### ISSUE TITLE
`azurerm_function_app` & `azurerm_function_app_slot` support for ip-restrictions scm/kudu site 

### ISSUE THREAD

**Issue by SebRosander** [terraform-providers/terraform-provider-azurerm#7431]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Add support for ip-restrictions scm/kudu site for function app & slot. 

### New or Affected Resource(s)

* azurerm_function_app
* azurerm_function_app_slot

### Potential Terraform Configuration

```hcl
  site_config {
   # Current network IP address restriction on function app
    ip_restriction {
      ip_address  = "1.2.3.4"
    }

   # Proposed new IP address restriction on function app SCM/KUDU site
    scm_ip_restriction {
      ip_address  = "5.6.7.8"
    }
  }
}
```

### References

* https://docs.microsoft.com/en-us/azure/app-service/app-service-ip-restrictions

* Same as  #3685 but for function app

* #6955

**Comment by alastairtree** [comment]
Thank you for this. The attached PR actually puts functions ahead of app service in terms of functionality because we still don't have IP Restrictions for SCM on app service slots yet.

**Comment by jackofallops** [comment]
closed by #7945 - released in 2.23.0
